### PR TITLE
Logistics various Improvements

### PR DESCRIPTION
**Pull request by tsh-odoo**
Description of the issue/feature this PR addresses:

Task : https://www.odoo.com/web#id=47933&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad ID : https://pad.odoo.com/p/r.b02f8a35aee224e9a211f0228d5aab90
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jir-odoo** [comment]
LGTM

**Comment by amoyaux** [comment]
@tsh-odoo Hello, could you rebase the branch please? I also think that you could remove the framework  commit and instead return 'mrp.act_mrp_product_produce' instead of windows close or reload.

**Comment by tsh-odoo** [comment]
> @tsh-odoo Hello, could you rebase the branch please? I also think that you could remove the framework commit and instead return 'mrp.act_mrp_product_produce' instead of windows close or reload.

@amoyaux Hi, I have resolved the conflicts and Branch is green now and here we need to reload the page(for both the cases 1: after few production user can discard, 2: after completion of all the production) to update consumed productlines. Thanks.

**Comment by amoyaux** [comment]
Hey @tsh-odoo I understand that you have to reload the view but we would like to avoid a patch in framework js without a task/tests. I thinks it possible to reload the views by returning an action containing the display of the wizard or the production order depending the button and the case, you can always override the discard button in order to do it.

**Comment by amoyaux** [comment]
Also could you split your commit in:
1 for produce wizard
1 for sale_stock
1 for maintenance
1 for purchase
1 for product
Do not forget to include a tilte for each commit and a description why the change has been made (not the change itself).

**Comment by amoyaux** [comment]
@tsh-odoo I did the last modification, here is my feedback:
1. Do not forget white space between tag and module name
2. Instead of explaining the change you did, explain the purpose of it.
The code was great.

Not really a feedback but just fyi,
I divided do_produce in 2 functions. _record_production private that do the modification on production and stock move. do_produce that call _record_production and then close the wizard.
I also replaced return {'type': 'ir.actions.client', 'tag': 'reload'} by return {'type': 'ir.actions.act_window_close'} since it will only reload the production view and not all the web client.

Thanks for the work done.

**Comment by amoyaux** [comment]
@sle-odoo you can r+

File: addons/mrp/wizard/mrp_product_produce.py [hunk 1]
```diff
@@ -121,6 +123,32 @@ def check_finished_move_lots(self):
                 pl.move_id._generate_consumed_move_line(pl.qty_done, self.lot_id, lot=pl.lot_id)
         return True
 
+    def _get_todo(self, production):
+        """ This method will return remaining todo quantity of production. """
```
Comment (sle-odoo) [inline addons/mrp/wizard/mrp_product_produce.py]: use self.production_id

File: addons/mrp/wizard/mrp_product_produce.py [hunk 2]
```diff
@@ -47,26 +44,31 @@ def default_get(self, fields):
...
+    @api.depends('product_qty')
+    def _compute_pending_production(self):
+        """ Compute if it exits remaining quantity once the quantity on the
+        current wizard will be processed. The purpose is to display or not
+        button 'continue'.
+        """
+        for product_produce in self:
+            main_product_moves = product_produce.production_id.move_finished_ids.filtered(lambda x: x.product_id.id == product_produce.production_id.product_id.id)
+            todo_quantity = product_produce.production_id.product_qty - (product_produce.product_qty + sum(main_product_moves.mapped('quantity_done')))
+            product_produce.is_pending_production = todo_quantity > 0.0
```
Comment (sle-odoo) [inline addons/mrp/wizard/mrp_product_produce.py]: maybe _get_todo - product_produce.product_qty ?

File: addons/mrp/wizard/mrp_product_produce.py (hunk 1 above)
Comment (amoyaux) [reply addons/mrp/wizard/mrp_product_produce.py]: Nope since used in default_get

File: addons/mrp/wizard/mrp_product_produce.py (hunk 2 above)
Comment (amoyaux) [reply addons/mrp/wizard/mrp_product_produce.py]: Done

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**